lib/Inst/Inst.cpp

```diff
@@ -77,6 +77,10 @@ bool Inst::operator<(const Inst &Other) const {
     return (*OpsA[I] < *OpsB[I]);
   }
 
+  if (HarvestKind != Other.HarvestKind)
+    return true;
```

Comment:
does this work? this just them them compare less than, but then the opposite comparison will also be less than.
don't you need to return something like (int)HarvestKind) < (int)Other.HarvestKind?

Reply:
Fixed in the new commit.